packages/extension/src/utils/decrypt-message.ts

```diff
@@ -0,0 +1,37 @@
+import { fromBase64, fromHex, fromUtf8 } from "@cosmjs/encoding";
+import { decrypt } from "eciesjs";
+import { store } from "../chatStore";
+
+export const decryptMessage = async (content: string, isSender: boolean) => {
+  try {
+    const state = store.getState();
+    const { user } = state;
+    const data = Buffer.from(content, "base64").toString("ascii");
+
+    //@ts-ignore "ignoring type checking here"
+    const dataEnvalopDecoded = JSON.parse(data);
+
+    const decodedData = fromUtf8(fromHex(dataEnvalopDecoded.data));
+
+    const msgrev = JSON.parse(decodedData);
+
+    const msgrevToAscii = msgrev.msgs[0].value.data;
+
+    const senTargetData = Buffer.from(fromBase64(msgrevToAscii)).toString("ascii");
+
+    const parsedData = JSON.parse(senTargetData);
+
+    const decryptedData = decrypt(
+      user.prvKey,
```

Comment:
Is this using the Fetch account private key to decrypt the message directly? 

I think we should strongly prefer doing some kind of key exchange to arrive at fixed-length "session keys" for symmetric encryption of messages. I think only using the Fetch account keys to encrypt/decrypt these fixed-length symmetric session keys is a much less risky proposition. In general I think we should try to avoid key re-use but especially between signing and encryption use cases.

Do we know what quality attributes are we expecting from the chat feature?